### PR TITLE
Add a basic feature grid

### PR DESCRIPTION
**Pull request by hwbllmnn**
Adds a simple table based on a GeoExt features store. The grid is editable, features are highlighted in the map on mouseover.

As opposed to the feature property grid it displays all features of a layer instead of just one.

@terrestris/devs please review

### CODE REVIEW

File: src/view/component/Map.js [hunk 1]
```diff
@@ -180,6 +180,23 @@ Ext.define('BasiGX.view.component.Map', {
...
         });
         map.addControl(attribution);
+    },
+
+    /**
+     * Retrieves a layer by its name property.
+     * @param  {String} name the name
+     * @return {ol.layer.Layer}     the layer, or undefined, if it doesn't exist
+     */
+    getLayerByName: function(name) {
```
Comment (weskamm) [inline src/view/component/Map.js]: i think the util package for map or layer should be more appropriate for this method

File: src/view/grid/FeatureGrid.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                })
+            }),
+            source: this.highlightSource
+        });
+        this.highlightLayer.set('name', 'highlight');
+        var trees = Ext.ComponentQuery.query('basigx-panel-legendtree');
+        var store;
+        if (trees.length > 0) {
+            store = trees[0].getStore();
+            store.suspendCollectionEvents();
```
Comment (weskamm) [inline src/view/grid/FeatureGrid.js]: maybe using a store filter is a safer way of ignoring this layer?

File: src/view/component/Map.js (hunk 1 above)
Comment (weskamm) [reply src/view/component/Map.js]: edit: getLayerBy should already do the job in the layer util

File: src/view/grid/FeatureGrid.js (hunk 2 above)
Comment (weskamm) [reply src/view/grid/FeatureGrid.js]: wait, i think this is even better:
https://github.com/terrestris/BasiGX/blob/14955933b6c4c85b9046a9d755dbf2ce358a87af/src/view/form/Print.js#L438-L441

File: src/view/grid/FeatureGrid.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    /**
+     * Extracts the feature schema from the first feature in the store.
+     * @param  {GeoExt.data.store.Features} store the layer store
+     * @return {Array}       an array with the feature attribute names
+     */
+    extractSchema: function(store) {
+        var columns = [];
+        var data = store.getData().items;
+        if (data.length > 0) {
+            Ext.iterate(store.getData().items[0].data, function(key, value) {
```
Comment (weskamm) [inline src/view/grid/FeatureGrid.js]: can replace `store.getData().items` with `data` here

**Comment by weskamm** [comment]
Added some minor comments..

File: src/view/component/Map.js (hunk 1 above)
Comment (hwbllmnn) [reply src/view/component/Map.js]: True, fixed.

File: src/view/grid/FeatureGrid.js (hunk 2 above)
Comment (hwbllmnn) [reply src/view/grid/FeatureGrid.js]: True, fixed.

File: src/view/grid/FeatureGrid.js (hunk 3 above)
Comment (hwbllmnn) [reply src/view/grid/FeatureGrid.js]: Done.

File: src/view/grid/FeatureGrid.js [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+            ptype: 'cellediting',
+            clicksToEdit: 1
+        }
+    }],
+
+    /**
+     *
+     */
+    initComponent: function() {
+        this.callParent([arguments]);
```
[the remark this answers is not in the document]
Comment (hwbllmnn) [reply src/view/grid/FeatureGrid.js]: Will be addressed in my upcoming merge-selection PR.

File: src/view/grid/FeatureGrid.js [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+        this.setLayerStore();
+        this.registerEvents();
+        var map = Ext.ComponentQuery.query('basigx-component-map')[0];
```
[the remark this answers is not in the document]
Comment (hwbllmnn) [reply src/view/grid/FeatureGrid.js]: Will be addressed in my upcoming merge-selection PR.

**Pull request merged**